ts/screens/wallet/payment/PickPaymentMethodScreen.tsx

```diff
@@ -48,108 +60,139 @@ type Props = ReturnType<typeof mapStateToProps> &
+                      paymentMethod={i.item}
+                      onPress={() =>
+                        props.navigateToConfirmOrPickPsp(
+                          // Since only credit cards are now accepted method we manage only this case
+                          // TODO: if payment methods different from credit card should be accepted manage every case
+                          convertWalletV2toWalletV1(i.item)
+                        )
+                      }
+                    />
+                  )}
                 />
-              )}
-            />
-            {wallets.length > 0 && <EdgeBorderComponent />}
-          </View>
-        </Content>
-
-        <View spacer={true} />
-
-        {wallets.some(myWallet => myWallet.creditCard?.brand === "AMEX") &&
-          verifica.importoSingoloVersamento >= amexPaymentWarningTreshold && (
-            <View style={styles.infoBoxContainer}>
-              <InfoBox alignedCentral={true} iconColor={IOColors.white}>
-                <Label weight={"Regular"} color="white">
-                  {I18n.t("wallet.alert.amex")}
-                </Label>
-              </InfoBox>
-            </View>
-          )}
-        <FooterWithButtons
-          type="TwoButtonsInlineThird"
-          leftButton={secondaryButtonProps}
-          rightButton={primaryButtonProps}
-        />
-      </BaseScreenComponent>
-    );
-  }
-}
+                <View spacer={true} />
+              </>
+            ) : (
+              <H4
+                weight={"Regular"}
+                color={"bluegreyDark"}
+                testID={"noWallets"}
+              >
+                {I18n.t("wallet.payWith.noWallets.text")}
+              </H4>
+            )}
+
+            {notPayableWallets.length > 0 && (
+              <>
+                <View spacer={true} />
+                <H4 color={"bluegreyDark"}>
+                  {I18n.t(
+                    "wallet.payWith.pickPaymentMethod.notAvailable.title"
+                  )}
+                </H4>
+                <View spacer={true} />
+                <FlatList
+                  testID={"notPayablePaymentMethodList"}
+                  removeClippedSubviews={false}
+                  data={notPayableWallets}
+                  keyExtractor={item => item.idWallet.toString()}
+                  ListFooterComponent={<View spacer />}
+                  renderItem={i => (
+                    <PickNotAvailablePaymentMethodListItem
+                      isFirst={i.index === 0}
+                      paymentMethod={i.item}
+                    />
+                  )}
+                />
+              </>
+            )}
+          </Content>
+        </ScrollView>
+        {renderFooterButtons(
+          props.navigateToTransactionSummary,
+          props.navigateToAddPaymentMethod
+        )}
+      </SafeAreaView>
+    </BaseScreenComponent>
+  );
+};
 
 const mapStateToProps = (state: GlobalState) => {
```

Comment:
What about this refactoring?

```typescript
function getValueOrEmptyArray(
  value: pot.Pot<ReadonlyArray<PaymentMethod>, unknown>
): ReadonlyArray<PaymentMethod> {
  return pot.getOrElse(value, []);
}

const mapStateToProps = (state: GlobalState) => {
  const potVisibleCreditCard = creditCardListVisibleInWalletSelector(state);
  const potVisibleBancomat = bancomatListVisibleInWalletSelector(state);
  const potVisibleBPay = bPayListVisibleInWalletSelector(state);
  const potVisibleSatispay = satispayListVisibleInWalletSelector(state);
  const potVisiblePrivative = privativeListVisibleInWalletSelector(state);
  const visibleWallets = [
    potVisibleCreditCard,
    potVisibleBancomat,
    potVisibleBPay,
    potVisibleSatispay,
    potVisiblePrivative
  ].reduce(
    (
      acc: ReadonlyArray<PaymentMethod>,
      curr: pot.Pot<ReadonlyArray<PaymentMethod>, unknown>
    ) => [...acc, ...getValueOrEmptyArray(curr)],
    [] as ReadonlyArray<PaymentMethod>
  );

  const potPsps = state.wallet.payment.psps;
  const isLoading =
    pot.isLoading(potVisibleCreditCard) || pot.isLoading(potPsps);

  return {
    // Considering that the creditCardListVisibleInWalletSelector return
    // all the visible credit card we need to filter them in order to extract
    // only the cards that can pay on IO.
    payableWallets: visibleWallets.filter(vPW => canMethodPay(vPW)),
    notPayableWallets: visibleWallets.filter(vPW => !canMethodPay(vPW)),
    isLoading,
    nameSurname: profileNameSurnameSelector(state)
  };
};
```

touched points

- `getValueOrEmptyArray` replace all `pot.getOrElse` (`map` that return the same value is useless)
-  user reduce instead of concat (it's my taste, I guess using `M` is overkilling for this scenario. Take which you prefer)
- `payableWallets` and `notPayableWallets` could be values instead of functions and they can be used as follows

`const { payableWallets, notPayableWallets } = props;`

Reply:
Really great readability! Thank you for the feedback!
Refactored in [2874f42](https://github.com/pagopa/io-app/pull/2926/commits/2874f42bdd4149703ae7ff6c87bbbf510d66bef5)